pkg/destroy/alibabacloud/alibabacloud.go

```diff
@@ -0,0 +1,1343 @@
+	return
+}
+
+func (o *ClusterUninstaller) listSecurityGroup(securityGroupIDs []string) (response *ecs.DescribeSecurityGroupsResponse, err error) {
+	request := ecs.CreateDescribeSecurityGroupsRequest()
+	securityGroupIDsString, err := json.Marshal(securityGroupIDs)
+	if err != nil {
+		return nil, err
+	}
+	request.SecurityGroupIds = string(securityGroupIDsString)
+	response, err = o.ecsClient.DescribeSecurityGroups(request)
+	return
+}
+
+func (o *ClusterUninstaller) listEcsInstance(instanceIDs []string) (response *ecs.DescribeInstancesResponse, err error) {
+	request := ecs.CreateDescribeInstancesRequest()
+	instanceIDsString, err := json.Marshal(instanceIDs)
+	if err != nil {
+		return nil, err
+	}
+	request.InstanceIds = string(instanceIDsString)
+	response, err = o.ecsClient.DescribeInstances(request)
+	return
+}
+
+func (o *ClusterUninstaller) modifyDeletionProtection(instanceID string) (err error) {
+	request := ecs.CreateModifyInstanceAttributeRequest()
+	request.InstanceId = instanceID
+	request.DeletionProtection = "false"
+	_, err = o.ecsClient.ModifyInstanceAttribute(request)
+	return
+}
+
+func (o *ClusterUninstaller) modifyECSInstancesDeletionProtection(instanceIDs []string) (err error) {
+	response, err := o.listEcsInstance(instanceIDs)
+	if err != nil {
+		return err
+	}
+	for _, instance := range response.Instances.Instance {
+		if instance.DeletionProtection {
+			err := o.modifyDeletionProtection(instance.InstanceId)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return
+}
+
+func (o *ClusterUninstaller) deleteEcsInstances() (err error) {
+	if len(o.TagResources.ecsInstances) <= 0 {
+		return nil
+	}
+
+	var instanceIDs []string
+	for _, instanceArn := range o.TagResources.ecsInstances {
+		instanceIDs = append(instanceIDs, instanceArn.ResourceID)
+	}
+
+	o.Logger.Debugf("Start to check and turn off ECS instances %q deletion protection", instanceIDs)
+	err = o.modifyECSInstancesDeletionProtection(instanceIDs)
+	if err != nil {
+		return err
+	}
+
+	o.Logger.Debugf("Start to delete ECS instances %q", instanceIDs)
+	request := ecs.CreateDeleteInstancesRequest()
+	request.InstanceId = &instanceIDs
+	request.Force = "true"
+	_, err = o.ecsClient.DeleteInstances(request)
+	if err != nil {
+		return err
+	}
+
+	err = wait.Poll(
+		5*time.Second,
+		5*time.Minute,
+		func() (bool, error) {
+			response, err := o.listEcsInstance(instanceIDs)
+			if err != nil {
+				return false, err
+			}
+			if response.TotalCount == 0 {
+				return true, nil
+			}
+			return false, nil
+		},
+	)
+	return
+}
+
+func (o *ClusterUninstaller) findResourcesByTag() (tagResources []tag.TagResource, err error) {
+	for _, tags := range o.Tags {
+		resources, err := o.ListTagResources(tags)
+		if err != nil {
+			return nil, err
+		}
+		tagResources = append(tagResources, resources...)
+	}
+	return tagResources, nil
+}
+
+func (o *ClusterUninstaller) ListTagResources(tags map[string]string) (tagResources []tag.TagResource, err error) {
```

Comment:
This is causing the go-lint test to fail: 

```
 /go/src/github.com/openshift/installer/pkg/destroy/alibabacloud/alibabacloud.go:987:1: exported method ClusterUninstaller.ListTagResources should have comment or be unexported
```

Reply:
done, please continue